tools/src/test/java/com/nedap/archie/archetypevalidator/ArchetypeValidatorTest.java

```diff
@@ -233,6 +233,22 @@ public void parentWithDifferentRmRelease() throws IOException, ADLParseException
         }
     }
 
+    @Test
+    public void infiniteSpecialisationTreeLoopTest() throws IOException, ADLParseException {
```

Comment:
Are there any other tests that should be added? Will it benefit on testing it with an infinite loop that only occurs after three archetypes:

archetype1 specialises archetype2
archetype2 specialised archetype3
archetype3 specialises archetype1 

??